test/cases/compile_errors/undefined_propagation_in_equality_operations.zig

```diff
@@ -0,0 +1,38 @@
+test "undefined propagation in equality operators for bool" {
+    var foo: bool = undefined;
+    _ = foo == undefined;
+    if (foo == undefined) {}
+}
+
+test "undefined propagation in equality operators for pointer" {
+    var foo: *i32 = undefined;
+    _ = foo == undefined;
+    if (foo == undefined) {}
+}
+
+test "undefined propagation in equality operators for optional" {
+    var foo: ?*i32 = undefined;
+    _ = foo == undefined;
+    if (foo == undefined) {}
+}
+
+test "undefined propagation in equality operators for error union" {
+    var foo: anyerror = undefined;
+    _ = foo == undefined;
```

Comment:
This looks like it is supposed to be instead the same expression from the if condition below.

Reply:
> This looks like it is supposed to be instead the same expression from the if condition below.

+